master/internal/db/postgres_checkpoints.go

```diff
@@ -0,0 +1,85 @@
+package db
+
+import (
+	"fmt"
+)
+
+// FilterForRegisteredCheckpoints gets the deleted checkpoints provided in the model registry.
+func (db *PgDB) FilterForRegisteredCheckpoints(deleteCheckpoints []string) ([]string, error) {
+	var checkpointIDRows []struct {
+		ID string
+	}
+
+	if err := db.queryRows(`
+	SELECT DISTINCT(c.uuid::text) AS ID FROM checkpoints_view AS c 
+	JOIN model_versions AS mv ON mv.checkpoint_uuid = c.uuid
+	WHERE c.uuid::text IN (SELECT UNNEST($1::text[])); 
+`, &checkpointIDRows, deleteCheckpoints); err != nil {
+		return nil, fmt.Errorf(
+			"querying for all requested delete checkpoints registered in model registry: %w", err)
+	}
+
+	var checkpointIDs []string
+
+	for _, cRow := range checkpointIDRows {
+		checkpointIDs = append(checkpointIDs, cRow.ID)
+	}
+
+	return checkpointIDs, nil
+}
+
+// MarkCheckpointsDeleted updates the provided delete checkpoints to DELETED state.
+func (db *PgDB) MarkCheckpointsDeleted(deleteCheckpoints []string) error {
+	_, err := db.sql.Exec(`UPDATE raw_checkpoints c
+    SET state = 'DELETED'
+    WHERE c.uuid::text IN (SELECT UNNEST($1::text[]))`, deleteCheckpoints)
+	if err != nil {
+		return fmt.Errorf("deleting checkpoints from raw_checkpoints: %w", err)
+	}
+
+	_, err = db.sql.Exec(`UPDATE checkpoints_v2 c
+    SET state = 'DELETED'
+    WHERE c.uuid::text IN (SELECT UNNEST($1::text[]))`, deleteCheckpoints)
+	if err != nil {
+		return fmt.Errorf("deleting checkpoints from checkpoints_v2: %w", err)
+	}
+
+	return nil
+}
+
+// ExperimentCheckpointGrouping represents a mapping of checkpoint uuids to experiment id.
+type ExperimentCheckpointGrouping struct {
+	EID          int
+	CUUIDsString string
```

Comment:
```suggestion
	ExperimentID          int
	CheckpointUUIDs  []string
```
[]string may just work, or you may need to use `pgxtype.StringArr`